bigdata_service/src/api/v1/posts.py

```
@@ -0,0 +1,150 @@
+async def _increase_views_transaction(ids: list[UUID]) -> list[Post]:
+    posts = []
+    """
+        Wrong way:
+        for id in ids:
+        post = await Post.find_one(Post.id == id)
+        post.views += 1
+        await post.replace()
+        posts.append(post)
+    """
+    async with await mongo.mongo_client.start_session() as session:
+        async with session.start_transaction():
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)